security/pkg/nodeagent/sds/server.go

```diff
@@ -81,12 +81,12 @@ func (s *Server) Stop() {
 	}
 }
 
-func (s *Server) initWorkloadSdsService(options *security.Options) {
+func (s *Server) initWorkloadSdsService() {
 	s.grpcWorkloadServer = grpc.NewServer(s.grpcServerOptions()...)
 	s.workloadSds.register(s.grpcWorkloadServer)
 
 	var err error
-	s.grpcWorkloadListener, err = uds.NewListener(options.WorkloadUDSPath)
+	s.grpcWorkloadListener, err = uds.NewListener(security.WorkloadIdentitySocketPath)
 	if err != nil {
```

Comment:
Can we make this `WorkloadIdentitySocketPath` configurable to make it more generic to meet the needs of other SDS?

Reply:
There is only one sds server for each pod which is no doubt, but my concern is the path names of uds: `/var/run/secrets/workload-spiffe-uds/socket` and cert: `/var/run/secrets/workload -spiffe-credentials` seems to be a bit specific. For HSM, we can reuse sockets in the same mount path, which is no problem, but I would like to suggest marking this path as a more generic path, such as `/var/run/ secrets/workload-external-uds/socket`, which seems more reasonable?

I think the best way is to make this name configurable by user who decide the name of the socket and secret, but no matter which option, each pod will only enable one SDS Server , such as Spire or HSM or other.

Reply:
Why? Its still spiffe credentials. Spiffe != Spire, and "Spiffe" and HSM or orthogonal. Spiffe just refers to the type of certificate.

Even if the name is not perfect, I would prefer to use an existing standard that is suboptimal than a new one

Reply:
And strongly prefer 'zero config' for users.

Reply:
I think we will have other cases that need rely on this external SDS server feature as well.   So it is better we make the path name more generic  like : `/var/run/secrets/external-sds/socket`  and  `/var/run/secrets/external-sds-credentials`   The current name is a little specific, right?

Reply:
Talked with @howardjohn since all cert will be in spiffe format,  agree it is a reasonable name  :)

Reply:
I also think the name is reasonable enough because we won't change the spiffe standard cert for Istio certificates, same as @irisdingbj, thanks @howardjohn : )